wordpress-mobile/WordPress-Android
# Add handling for notifications deeplinks

Fixes #14450 

This PR adds handling for the notifications links. Both the email and the direct links are now opened on the notifications tab. You might notice that if you open the links in the browser, they take you to the `/me/notifications/` tab which is notifications settings. I think that's because there is no notifications screen on the web. I've checked with iOS and they handle these links the same way it's implemented in this PR.

To test:
- Click on the following email link `https://public-api.wordpress.com/mbar/?stat=groovemails-events&bin=wpcom_email_click&redirect_to=https://wordpress.com/notifications/`
- Notice the app opens in the main activity with the notifications screen open

To test:
- Click on the following email link `https://wordpress.com/notifications/`
- Notice the app opens in the main activity with the notifications screen open


## Regression Notes
1. Potential unintended areas of impact
- deeplinks

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- unit tests and manual testing

3. What automated tests I added (or what prevented me from doing so)
- unit tests for the view model and the notifications link handler

PR submission checklist:
